In [ ]:
from fastai.tabular.all import *
from fastai.collab import *
from fastai.data.transforms import *
import torch
import numpy as np
import pandas as pd
from sklearn.model_selection import ParameterGrid
from functools import partial

Developing a recommendation system for a **NO-Q** app, we were focusing on two major scenarios: new users and users who have purchased and rated products in the past.
For users who have purchased and rated products previously, we use their purchase history and ratings to generate personalized recommendations based on collaborative filtering.

**Data**

In user-based filtering, the algorithm identifies users with similar preferences and recommends items that these users have previously enjoyed. To illustrate this approach, we generated a dataset containing 700 users and 441 items rated by users on a scale from 0 to 1 (like/dislike)

The dataset is generated in such a way that some users rated many items while most rated only a few, with some items remaining unrated - a realistic representation of a typical data

In [ ]:
#Load data
df = pd.read_csv('User ratings.csv')
cocktail_strings = df['title'].unique()

In [ ]:
# Create mapping of cocktail strings to integer IDs
cocktail_id_map = {s: i for i, s in enumerate(cocktail_strings)}

In [ ]:
# Replace cocktail strings with their IDs in the dataset
df['cocktail'] = df['title'].map(cocktail_id_map)
df = df[['user','cocktail','rating','title']]
ratings = df

**Optimization with grid search**

The hyperparameters being varied here include learning rate ('lr'), weight decay ('wd'), number of factors ('n_factors'), batch size ('bs'), number of epochs ('epochs'), and number of layers in the neural network ('layers')

In [ ]:
# Define hyperparameter grid
param_grid = {
    'lr': [1e-3, 5e-3, 1e-2],
    'wd': [0.01, 0.1, 0.2],
    'n_factors': [20, 50, 100],
    'bs': [32, 64, 128],
    'epochs': [3, 5],
    'layers': [
        [100, 50],
        [200, 100],
        [110,100],
    ]
}

In [ ]:
# Function to train model with given hyperparameters and return accuracy
def train_model(ratings, param, seed=42):
    torch.manual_seed(seed)
    dls = CollabDataLoaders.from_df(ratings, item_name='title', bs=param['bs'])
    learn = collab_learner(dls, n_factors=param['n_factors'], y_range=(0, 1.0), layers=param['layers'], loss_func=BCEWithLogitsLossFlat())
    learn.fit_one_cycle(param['epochs'], param['lr'], wd=param['wd'])
    
    preds, targets = learn.get_preds()
    mse = ((preds - targets)**2).mean()
    return mse

In [ ]:
# Grid search
best_mse = 1
best_params = None
for param_set in ParameterGrid(param_grid):
    mse = train_model(ratings, param_set)
    print(f"Parameters: {param_set}, MSE: {mse}")
    if mse < best_mse:
        best_mse = mse
        best_params = param_set

print(f"Best Parameters: {best_params}, Best MSE: {best_mse}")

Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2598572075366974


Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2598685622215271


Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 20, 'wd': 0.2}, MSE: 0.25985637307167053


Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2598913908004761


Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2598811686038971


Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 50, 'wd': 0.2}, MSE: 0.25989392399787903


Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 100, 'wd': 0.01}, MSE: 0.25988101959228516


Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 100, 'wd': 0.1}, MSE: 0.25988104939460754


Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 100, 'wd': 0.2}, MSE: 0.2598865330219269


Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 20, 'wd': 0.01}, MSE: 0.25885361433029175


Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2589115798473358


Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2589307129383087


Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2588590979576111


Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2589628994464874


Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2590164542198181


Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 100, 'wd': 0.01}, MSE: 0.25889042019844055


Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 100, 'wd': 0.1}, MSE: 0.2589706778526306


Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 100, 'wd': 0.2}, MSE: 0.2590266466140747


Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 20, 'wd': 0.01}, MSE: 0.25797829031944275


Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2581688463687897


Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 20, 'wd': 0.2}, MSE: 0.25830888748168945


Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2585093677043915


Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2584352195262909


Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2585429549217224


Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2591160535812378


Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 100, 'wd': 0.1}, MSE: 0.2589970529079437


Parameters: {'bs': 32, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 100, 'wd': 0.2}, MSE: 0.25889265537261963


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.01}, MSE: 0.25986039638519287


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.1}, MSE: 0.25986477732658386


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2598673105239868


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2598819136619568


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.1}, MSE: 0.25989192724227905


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.2}, MSE: 0.25988486409187317


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2598712146282196


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.1}, MSE: 0.2598825991153717


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.2}, MSE: 0.25987258553504944


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.01}, MSE: 0.25877678394317627


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2588966488838196


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.2}, MSE: 0.25897273421287537


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.01}, MSE: 0.25889596343040466


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2589159905910492


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2589697241783142


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2589360773563385


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.1}, MSE: 0.25888967514038086


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.2}, MSE: 0.25900787115097046


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2580638527870178


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2581019103527069


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2582303583621979


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2582755386829376


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.1}, MSE: 0.25839200615882874


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.2}, MSE: 0.25839975476264954


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2588975131511688


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.1}, MSE: 0.25900566577911377


Parameters: {'bs': 32, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.2}, MSE: 0.2588684558868408


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2598628103733063


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2598547041416168


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2598626911640167


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2598731815814972


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.1}, MSE: 0.259893536567688


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.2}, MSE: 0.25988420844078064


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.01}, MSE: 0.25986889004707336


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.1}, MSE: 0.25987914204597473


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.2}, MSE: 0.25988394021987915


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.01}, MSE: 0.25881707668304443


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.1}, MSE: 0.25891709327697754


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2589553892612457


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.01}, MSE: 0.25884708762168884


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2589494585990906


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2590332627296448


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2589362859725952


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.1}, MSE: 0.2589763104915619


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.2}, MSE: 0.2589637339115143


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2580467164516449


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2581559717655182


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2583024501800537


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.01}, MSE: 0.25833654403686523


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2582961916923523


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2584676146507263


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2585947513580322


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.1}, MSE: 0.2588573694229126


Parameters: {'bs': 32, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.2}, MSE: 0.25871196389198303


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 20, 'wd': 0.01}, MSE: 0.25967171788215637


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2596694231033325


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2596855163574219


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 50, 'wd': 0.01}, MSE: 0.25969672203063965


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2596933841705322


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 50, 'wd': 0.2}, MSE: 0.25970619916915894


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2596794366836548


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 100, 'wd': 0.1}, MSE: 0.25968700647354126


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 100, 'wd': 0.2}, MSE: 0.25969836115837097


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 20, 'wd': 0.01}, MSE: 0.25821441411972046


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 20, 'wd': 0.1}, MSE: 0.25830966234207153


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 20, 'wd': 0.2}, MSE: 0.25844141840934753


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 50, 'wd': 0.01}, MSE: 0.25854063034057617


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 50, 'wd': 0.1}, MSE: 0.25849488377571106


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2586614191532135


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 100, 'wd': 0.01}, MSE: 0.258884996175766


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 100, 'wd': 0.1}, MSE: 0.2588689923286438


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 100, 'wd': 0.2}, MSE: 0.25894057750701904


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2584729492664337


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 20, 'wd': 0.1}, MSE: 0.25832751393318176


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2581300735473633


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2599026560783386


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 50, 'wd': 0.1}, MSE: 0.259416788816452


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 50, 'wd': 0.2}, MSE: 0.25918981432914734


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 100, 'wd': 0.01}, MSE: 0.26109984517097473


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 100, 'wd': 0.1}, MSE: 0.26028746366500854


Parameters: {'bs': 32, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 100, 'wd': 0.2}, MSE: 0.2600240111351013


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2596604824066162


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.1}, MSE: 0.25967299938201904


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2596845328807831


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2596932649612427


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.1}, MSE: 0.259690523147583


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2597052752971649


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2596825063228607


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.1}, MSE: 0.25968465209007263


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.2}, MSE: 0.25969937443733215


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.01}, MSE: 0.25816741585731506


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.1}, MSE: 0.25831761956214905


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2584298849105835


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.01}, MSE: 0.25849583745002747


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.1}, MSE: 0.25849539041519165


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.2}, MSE: 0.25859346985816956


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2588523328304291


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.1}, MSE: 0.25882187485694885


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.2}, MSE: 0.25882112979888916


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.01}, MSE: 0.25868648290634155


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2582418620586395


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2582160532474518


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.01}, MSE: 0.26002809405326843


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2593359351158142


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.2}, MSE: 0.25897449254989624


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.01}, MSE: 0.26049041748046875


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.1}, MSE: 0.26089224219322205


Parameters: {'bs': 32, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.2}, MSE: 0.26014575362205505


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2596687376499176


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2596760392189026


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2596791982650757


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2596878111362457


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.1}, MSE: 0.25969892740249634


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.2}, MSE: 0.25970879197120667


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2596795856952667


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.1}, MSE: 0.25968411564826965


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.2}, MSE: 0.25969985127449036


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.01}, MSE: 0.25823134183883667


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2582673132419586


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2584668695926666


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.01}, MSE: 0.25841760635375977


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.1}, MSE: 0.25845998525619507


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2586231529712677


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2590719759464264


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.1}, MSE: 0.25890102982521057


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.2}, MSE: 0.2588469088077545


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.01}, MSE: 0.25864142179489136


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.1}, MSE: 0.25825268030166626


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.2}, MSE: 0.25817736983299255


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.01}, MSE: 0.26013004779815674


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.1}, MSE: 0.25932881236076355


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.2}, MSE: 0.259081095457077


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2608046531677246


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.1}, MSE: 0.260561466217041


Parameters: {'bs': 32, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.2}, MSE: 0.26005855202674866


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2599385976791382


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2599391043186188


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 20, 'wd': 0.2}, MSE: 0.25993961095809937


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 50, 'wd': 0.01}, MSE: 0.25995829701423645


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 50, 'wd': 0.1}, MSE: 0.25994816422462463


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 50, 'wd': 0.2}, MSE: 0.25996312499046326


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 100, 'wd': 0.01}, MSE: 0.25994694232940674


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 100, 'wd': 0.1}, MSE: 0.25994956493377686


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 100, 'wd': 0.2}, MSE: 0.25995510816574097


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2591482400894165


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2591630518436432


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2591879665851593


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 50, 'wd': 0.01}, MSE: 0.25917521119117737


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2591601312160492


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2592245936393738


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 100, 'wd': 0.01}, MSE: 0.25918036699295044


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 100, 'wd': 0.1}, MSE: 0.2591998279094696


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 100, 'wd': 0.2}, MSE: 0.2592236399650574


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2582865059375763


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2584761381149292


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2584720849990845


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 50, 'wd': 0.01}, MSE: 0.25839415192604065


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 50, 'wd': 0.1}, MSE: 0.25844883918762207


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 50, 'wd': 0.2}, MSE: 0.25848615169525146


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2585648000240326


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 100, 'wd': 0.1}, MSE: 0.2586384415626526


Parameters: {'bs': 64, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 100, 'wd': 0.2}, MSE: 0.2586513161659241


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2599339485168457


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2599354386329651


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.2}, MSE: 0.25993478298187256


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.01}, MSE: 0.25995054841041565


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2599497437477112


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2599555552005768


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.01}, MSE: 0.25994428992271423


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.1}, MSE: 0.25994718074798584


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.2}, MSE: 0.2599470317363739


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.01}, MSE: 0.25917431712150574


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.1}, MSE: 0.25918272137641907


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.2}, MSE: 0.25918710231781006


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2591829001903534


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.1}, MSE: 0.25921744108200073


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.2}, MSE: 0.25922003388404846


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2591962218284607


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.1}, MSE: 0.25920942425727844


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.2}, MSE: 0.2592132091522217


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.01}, MSE: 0.25834253430366516


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.1}, MSE: 0.25841057300567627


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2585177719593048


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.01}, MSE: 0.25844091176986694


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2584545910358429


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.2}, MSE: 0.25860947370529175


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2585413157939911


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.1}, MSE: 0.25860801339149475


Parameters: {'bs': 64, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.2}, MSE: 0.25859856605529785


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2599377930164337


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.1}, MSE: 0.25993627309799194


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2599402964115143


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.01}, MSE: 0.25995877385139465


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.1}, MSE: 0.25995659828186035


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2599624991416931


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2599492073059082


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.1}, MSE: 0.2599504292011261


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.2}, MSE: 0.25994908809661865


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2591521441936493


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.1}, MSE: 0.25914961099624634


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2591749429702759


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2591836452484131


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.1}, MSE: 0.259209543466568


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2592118978500366


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.01}, MSE: 0.259136825799942


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.1}, MSE: 0.25917720794677734


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.2}, MSE: 0.25921356678009033


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2583255469799042


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.1}, MSE: 0.25838255882263184


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2585407495498657


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2584080696105957


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2584458589553833


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.2}, MSE: 0.25858020782470703


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2584986388683319


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.1}, MSE: 0.2585836946964264


Parameters: {'bs': 64, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.2}, MSE: 0.2586166560649872


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 20, 'wd': 0.01}, MSE: 0.25978896021842957


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2597878575325012


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2597920894622803


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 50, 'wd': 0.01}, MSE: 0.25980690121650696


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 50, 'wd': 0.1}, MSE: 0.259815514087677


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2598147988319397


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2598037123680115


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 100, 'wd': 0.1}, MSE: 0.25980353355407715


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 100, 'wd': 0.2}, MSE: 0.25980570912361145


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2585502564907074


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 20, 'wd': 0.1}, MSE: 0.25861120223999023


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2586508095264435


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2586229145526886


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 50, 'wd': 0.1}, MSE: 0.25864648818969727


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 50, 'wd': 0.2}, MSE: 0.25873515009880066


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2586652934551239


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 100, 'wd': 0.1}, MSE: 0.25875890254974365


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 100, 'wd': 0.2}, MSE: 0.2587791681289673


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2578991949558258


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 20, 'wd': 0.1}, MSE: 0.25791794061660767


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2581043541431427


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 50, 'wd': 0.01}, MSE: 0.25886836647987366


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 50, 'wd': 0.1}, MSE: 0.25878357887268066


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2586742043495178


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 100, 'wd': 0.01}, MSE: 0.25974035263061523


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 100, 'wd': 0.1}, MSE: 0.25968360900878906


Parameters: {'bs': 64, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 100, 'wd': 0.2}, MSE: 0.259457528591156


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.01}, MSE: 0.25978317856788635


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2597947418689728


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2597951292991638


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2598089873790741


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.1}, MSE: 0.259811669588089


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2598154544830322


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2597978115081787


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.1}, MSE: 0.259798526763916


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.2}, MSE: 0.25980526208877563


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2585136592388153


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2585705518722534


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.2}, MSE: 0.25866666436195374


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2585820257663727


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.1}, MSE: 0.25861695408821106


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.2}, MSE: 0.25870481133461


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2586984932422638


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.1}, MSE: 0.2587067484855652


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.2}, MSE: 0.2587772011756897


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2578418552875519


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.1}, MSE: 0.25795531272888184


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.2}, MSE: 0.25793635845184326


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.01}, MSE: 0.25906556844711304


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2588094174861908


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.2}, MSE: 0.25877413153648376


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.01}, MSE: 0.25978752970695496


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.1}, MSE: 0.2595365345478058


Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.2}, MSE: 0.2595195472240448


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.01}, MSE: 0.25978562235832214


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.1}, MSE: 0.25978797674179077


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.2}, MSE: 0.25979360938072205


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.01}, MSE: 0.25980180501937866


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2598152160644531


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2598108649253845


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2598022520542145


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.1}, MSE: 0.2598016560077667


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.2}, MSE: 0.25980496406555176


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.01}, MSE: 0.25855395197868347


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.1}, MSE: 0.25858962535858154


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2586510479450226


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2586297392845154


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.1}, MSE: 0.25869813561439514


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.2}, MSE: 0.25871989130973816


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.01}, MSE: 0.25870344042778015


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.1}, MSE: 0.25877195596694946


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.2}, MSE: 0.2587924003601074


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2580316662788391


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.1}, MSE: 0.25806859135627747


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.2}, MSE: 0.25812482833862305


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.01}, MSE: 0.25876814126968384


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.1}, MSE: 0.258730411529541


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2585786283016205


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.01}, MSE: 0.25969207286834717


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.1}, MSE: 0.25975459814071655


Parameters: {'bs': 64, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.2}, MSE: 0.2593662738800049


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2599981427192688


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2599968910217285


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 20, 'wd': 0.2}, MSE: 0.25999459624290466


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2600201666355133


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2600218653678894


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2600141763687134


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2600148022174835


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 100, 'wd': 0.1}, MSE: 0.26001429557800293


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 100, 'wd': 0.2}, MSE: 0.26001402735710144


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2594500780105591


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 20, 'wd': 0.1}, MSE: 0.25942862033843994


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2594447433948517


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2594708204269409


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 50, 'wd': 0.1}, MSE: 0.25945600867271423


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 50, 'wd': 0.2}, MSE: 0.25947803258895874


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2594427466392517


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 100, 'wd': 0.1}, MSE: 0.2594757676124573


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 100, 'wd': 0.2}, MSE: 0.2594341039657593


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 20, 'wd': 0.01}, MSE: 0.25877007842063904


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2588152587413788


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 20, 'wd': 0.2}, MSE: 0.25886693596839905


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2587840259075165


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 50, 'wd': 0.1}, MSE: 0.25878795981407166


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2588868737220764


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 100, 'wd': 0.01}, MSE: 0.25882840156555176


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 100, 'wd': 0.1}, MSE: 0.2588202953338623


Parameters: {'bs': 128, 'epochs': 3, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 100, 'wd': 0.2}, MSE: 0.25887978076934814


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.01}, MSE: 0.25999966263771057


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2600032091140747


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2599964439868927


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2600163519382477


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2600143551826477


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.2}, MSE: 0.26002055406570435


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.01}, MSE: 0.26001232862472534


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.1}, MSE: 0.26001492142677307


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.2}, MSE: 0.2600114941596985


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2594262659549713


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2594453692436218


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2594594359397888


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.01}, MSE: 0.259451299905777


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2594776749610901


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.2}, MSE: 0.25947776436805725


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.01}, MSE: 0.25944486260414124


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.1}, MSE: 0.2594430148601532


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.2}, MSE: 0.259471595287323


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.01}, MSE: 0.25879552960395813


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.1}, MSE: 0.25878801941871643


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2588583528995514


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2588261365890503


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2588634192943573


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.2}, MSE: 0.25884756445884705


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2588335871696472


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.1}, MSE: 0.2588556706905365


Parameters: {'bs': 128, 'epochs': 3, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.2}, MSE: 0.25886592268943787


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2600064277648926


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2599959671497345


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.2}, MSE: 0.25999578833580017


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2600196897983551


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2600195109844208


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2600187063217163


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2600122392177582


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.1}, MSE: 0.2600110173225403


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.2}, MSE: 0.2600121796131134


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2594306468963623


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.1}, MSE: 0.25944381952285767


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.2}, MSE: 0.25945132970809937


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2594659626483917


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2594750225543976


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2594912350177765


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.01}, MSE: 0.25945714116096497


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.1}, MSE: 0.2594638466835022


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.2}, MSE: 0.25947776436805725


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.01}, MSE: 0.25881683826446533


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2588174045085907


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2588827908039093


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2588440775871277


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2588401436805725


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2588956952095032


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.01}, MSE: 0.25882071256637573


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.1}, MSE: 0.2588439881801605


Parameters: {'bs': 128, 'epochs': 3, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.2}, MSE: 0.25885799527168274


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2598936855792999


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2598947286605835


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 20, 'wd': 0.2}, MSE: 0.25989311933517456


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2599135935306549


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2599110007286072


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2599160075187683


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 100, 'wd': 0.01}, MSE: 0.25990551710128784


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 100, 'wd': 0.1}, MSE: 0.25990769267082214


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.001, 'n_factors': 100, 'wd': 0.2}, MSE: 0.2599051892757416


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2589639723300934


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 20, 'wd': 0.1}, MSE: 0.25899538397789


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 20, 'wd': 0.2}, MSE: 0.25901302695274353


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2589849531650543


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2590212821960449


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2590358555316925


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 100, 'wd': 0.01}, MSE: 0.25897687673568726


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 100, 'wd': 0.1}, MSE: 0.25897225737571716


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.005, 'n_factors': 100, 'wd': 0.2}, MSE: 0.2590268850326538


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2580537497997284


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2581245005130768


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2582257091999054


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 50, 'wd': 0.01}, MSE: 0.25812992453575134


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2582281827926636


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2583064138889313


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 100, 'wd': 0.01}, MSE: 0.258476585149765


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 100, 'wd': 0.1}, MSE: 0.2584623098373413


Parameters: {'bs': 128, 'epochs': 5, 'layers': [100, 50], 'lr': 0.01, 'n_factors': 100, 'wd': 0.2}, MSE: 0.25847768783569336


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2598951458930969


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2598959803581238


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2598952054977417


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2599136233329773


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2599113881587982


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2599143087863922


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.01}, MSE: 0.259905606508255


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.1}, MSE: 0.2599062919616699


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.2}, MSE: 0.2599058151245117


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.01}, MSE: 0.25893306732177734


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2589796185493469


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.2}, MSE: 0.25900259613990784


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.01}, MSE: 0.25897911190986633


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2590081989765167


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2590333819389343


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.01}, MSE: 0.25898799300193787


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.1}, MSE: 0.2589988112449646


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.2}, MSE: 0.2590101361274719


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2580468952655792


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2581155002117157


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.2}, MSE: 0.25816982984542847


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2582205533981323


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.1}, MSE: 0.25822728872299194


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.2}, MSE: 0.25830700993537903


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.01}, MSE: 0.25846800208091736


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.1}, MSE: 0.258463591337204


Parameters: {'bs': 128, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.2}, MSE: 0.2584832012653351


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.01}, MSE: 0.25989043712615967


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.1}, MSE: 0.2598959803581238


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2598969042301178


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.01}, MSE: 0.25990980863571167


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2599155604839325


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2599116563796997


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.01}, MSE: 0.2599077522754669


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.1}, MSE: 0.25990599393844604


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.001, 'n_factors': 100, 'wd': 0.2}, MSE: 0.25990918278694153


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.01}, MSE: 0.2589682638645172


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.1}, MSE: 0.25898805260658264


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 20, 'wd': 0.2}, MSE: 0.25900277495384216


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.01}, MSE: 0.2589854300022125


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.1}, MSE: 0.2590009570121765


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2590397298336029


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.01}, MSE: 0.25896793603897095


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.1}, MSE: 0.259014755487442


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.005, 'n_factors': 100, 'wd': 0.2}, MSE: 0.2590564489364624


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.01}, MSE: 0.25801265239715576


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.1}, MSE: 0.25808897614479065


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.2}, MSE: 0.2581970989704132


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.01}, MSE: 0.25824999809265137


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.1}, MSE: 0.25825613737106323


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 50, 'wd': 0.2}, MSE: 0.2583203911781311


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.01}, MSE: 0.25845661759376526


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.1}, MSE: 0.2584664225578308


Parameters: {'bs': 128, 'epochs': 5, 'layers': [110, 100], 'lr': 0.01, 'n_factors': 100, 'wd': 0.2}, MSE: 0.258533775806427
Best Parameters: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.01}, Best MSE: 0.2578418552875519


In [ ]:
# Train the model with the best hyperparameters
best_dls = CollabDataLoaders.from_df(ratings, item_name='title', bs=best_params['bs'])
best_learn = collab_learner(best_dls, n_factors=best_params['n_factors'], layers = best_params['layers'], y_range=(0, 1.0), loss_func=BCEWithLogitsLossFlat())
best_learn.fit_one_cycle(best_params['epochs'], best_params['lr'], wd=best_params['wd'])


**Best Parameters**: {'bs': 64, 'epochs': 5, 'layers': [200, 100], 'lr': 0.01, 'n_factors': 20, 'wd': 0.01}, Best MSE: 0.2578418552875519

**The Mean Squared Error (MSE)** metric is commonly used in collaborative filtering to evaluate the accuracy of predicted ratings. It measures the average of the squared differences between the predicted and actual ratings for a set of users and items. An MSE value of 0 indicates a perfect match between the predicted and actual ratings, while higher values indicate larger prediction errors. An MSE value of around 0.25 is generally considered acceptable in collaborative filtering, as it indicates a reasonable level of accuracy in the predicted ratings. 

In [ ]:

# Menu
items_df = pd.DataFrame({'user': 10,
                      'title': ['Bluebird', 'Boston Sour', 'Irish Spring', 'Zipperhead', 'ACID', 'Spritz', 'Kir Royale', 'Affair', 'Lord and Lady', 'Van Vleet']})  # positions on the menu


In [ ]:

# Learner to get predictions for the items (the probability of liking the item)
preds, _ = best_learn.get_preds(dl=best_dls.test_dl(items_df))

# Get the index of the item with the highest predicted probability
most_likely_item_idx = np.argmax(preds)
most_likely_item_idx = most_likely_item_idx.item()
most_likely_item_title = items_df.loc[most_likely_item_idx, 'title']
print(f"Recommended for you: {most_likely_item_title}")

g = ratings.groupby('title')['rating'].count()
top_cocktails = g.sort_values(ascending=False).index.values[:1000]
top_cocktails[:10]

cocktail_bias = best_learn.model.bias(top_cocktails, is_item=True)
cocktail_bias.shape

mean_ratings = ratings.groupby('title')['rating'].mean()
cocktail_ratings = [(b, i, mean_ratings.loc[i]) for i, b in zip(top_cocktails, cocktail_bias)]

item0 = lambda o: o[0]
items_titles = items_df['title'].unique()

# Filter cocktail_ratings to only include ratings for items on the menu
filtered_cocktail_ratings = [(b, i, mean_ratings.loc[i]) for b, i, r in cocktail_ratings if i in items_titles]
list = sorted(filtered_cocktail_ratings, key=lambda o: o[0], reverse=True)[0:1]
print(f"Top drink on the menu: {list[0][1]}")

Recommended for you: Irish Spring
Top drink on the menu: Zipperhead
